streams/src/test/java/org/apache/kafka/streams/processor/internals/GlobalStreamThreadTest.java

```diff
@@ -98,7 +98,7 @@ public String newStoreName(final String prefix) {
             null,
             GLOBAL_STORE_TOPIC_NAME,
             "processorName",
-            () -> ProcessorAdapter.adapt(new KTableSource<>(GLOBAL_STORE_NAME, GLOBAL_STORE_NAME).get()));
+            () -> new KTableSource<>(GLOBAL_STORE_NAME, GLOBAL_STORE_NAME).get());
```

Comment:
I was surprised that only this test was driving that builder change, so I went looking for the usages in production code. If you look at GlobalStoreNode, you'll see that it doesn't use KTableSource, but instead just takes a `ProcessorSupplier<KIn, VIn, Void, Void>`.

In other words, this is a case where the new, stricter, API is revealing a problem with our code, not the other way around.

Can we just write a quick ProcessorSupplier implementation here suitable to make the test pass?

Reply:
Oh! great catch! You're right, I've rolled back the change and test is working.